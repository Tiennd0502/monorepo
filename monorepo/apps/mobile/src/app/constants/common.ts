export const CONGRATS_IMAGE = 'https://i.ibb.co/nkf6hFr/congrats.png';
