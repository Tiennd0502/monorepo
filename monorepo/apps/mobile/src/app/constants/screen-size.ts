import { Dimensions } from 'react-native';

export const BASE_SCREEN_WIDTH = 412;
export const BASE_SCREEN_HEIGHT = 842;

export const SCREEN_WIDTH: number = Dimensions.get('window').width;
export const SCREEN_HEIGHT: number = Dimensions.get('window').height;

export const CONTENT_WIDTH: number = SCREEN_WIDTH - 40;
