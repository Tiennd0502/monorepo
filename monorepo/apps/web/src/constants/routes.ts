export const ROUTES = {
  HOME: '/',
  VERIFY_OTP: '/verify-otp',
  SIGN_UP: '/sign-up',
  LOGIN: '/login',
};
