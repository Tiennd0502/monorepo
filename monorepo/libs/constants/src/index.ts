export * from './apis';
export * from './message';
export * from './commons';
export * from './product';
export * from './category';
export * from './validation';
