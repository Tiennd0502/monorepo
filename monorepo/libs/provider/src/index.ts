export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
