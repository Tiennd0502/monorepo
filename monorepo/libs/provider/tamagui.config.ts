import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
