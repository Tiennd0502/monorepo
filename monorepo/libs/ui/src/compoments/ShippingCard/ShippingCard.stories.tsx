import { Meta, StoryObj } from '@storybook/react';
import { Stack } from 'tamagui';

import ShippingCard from '.';
import { SHIPPING_ADDRESS } from '@monorepo/mocks';

const meta: Meta<typeof ShippingCard> = {
  title: 'Components/ShippingCard',
  component: ShippingCard,
  decorators: [
    (Story) => (
      <Stack width={450}>
        <Story />
      </Stack>
    ),
  ],
};

export default meta;

type Story = StoryObj<typeof ShippingCard>;

export const Default: Story = {
  args: {
    item: SHIPPING_ADDRESS[0],
    onEdit: () => null,
  },
};

export const NoEdit: Story = {
  args: {
    item: SHIPPING_ADDRESS[0],
    onEdit: undefined,
  },
};
