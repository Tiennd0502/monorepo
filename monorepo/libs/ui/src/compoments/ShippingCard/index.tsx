import { memo, useCallback } from 'react';
import { Stack, XStack } from 'tamagui';

// Types
import { Shipping } from '@monorepo/types';

// Components
import Text from '../Text';
import Divider from '../Divider';
import { EditIcon } from '../icons';
import { shadows } from '../../themes';

interface ShippingCardProps {
  item: Shipping;
  onEdit?: (value: string) => void;
}

const ShippingCard = ({
  item: { id, name, address },
  onEdit,
}: ShippingCardProps) => {
  const handleEditClick = useCallback(() => {
    onEdit && onEdit(id);
  }, [id, onEdit]);

  return (
    <Stack
      style={shadows.card}
      borderRadius="$2"
      backgroundColor="$backgroundSecondary"
    >
      <XStack
        justifyContent="space-between"
        paddingVertical={'$3.75'}
        paddingLeft="$5"
        paddingRight={'$3.75'}
      >
        <Text color="$textPrimary" size="large" bold>
          {name}
        </Text>
        {onEdit && <EditIcon onPress={handleEditClick} />}
      </XStack>
      <Divider height={onEdit ? '$px' : '$0.5'} color="$backgroundTertiary" />
      <Text
        paddingHorizontal="$5"
        paddingTop="$3"
        paddingBottom="$4.5"
        color="$textLabel"
      >
        {address}
      </Text>
    </Stack>
  );
};

export default memo(ShippingCard);
