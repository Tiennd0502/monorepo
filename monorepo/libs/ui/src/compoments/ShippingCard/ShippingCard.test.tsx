import { fireEvent, render, waitFor } from '@testing-library/react-native';

import { SHIPPING_ADDRESS } from '@monorepo/constants';
import { TamaguiProvider } from '@monorepo/provider';
import ShippingCard from '.';

const onEdit = jest.fn();

const baseProps = {
  item: SHIPPING_ADDRESS[0],
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <ShippingCard {...props} />
    </TamaguiProvider>
  );
};


describe('ShippingCard', () => {

  it('should render ShippingCard component', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('should render ShippingCard component with onEdit props', () => {
    const container = setup({ onEdit });

    expect(container).toMatchSnapshot();
  });

  it('calls onEdit when icon edit clicked', () => {
    const {getByTestId} = setup({ onEdit });
    waitFor(() => {
      const editIcon = getByTestId('edit-icon');
      fireEvent.press(editIcon);

      expect(onEdit).toHaveBeenCalled();
    });
  });
});
