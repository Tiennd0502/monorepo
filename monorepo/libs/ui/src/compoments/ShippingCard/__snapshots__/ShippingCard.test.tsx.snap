// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingCard should render ShippingCard component 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "elevation": 16,
      "shadowColor": "$boxShadowSecondary",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 4,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 15,
        "paddingLeft": 20,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#242424",
          "fontFamily": "Inter, Helvetica, Arial, sans-serif",
          "fontSize": 18,
          "fontWeight": "bold",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      Bruno Fernandes
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#f0f0f0",
        "height": 2,
        "width": "100%",
      }
    }
  />
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#808080",
        "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
        "fontSize": 14,
        "fontWeight": "normal",
        "lineHeight": 20,
        "paddingBottom": 18,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 12,
      }
    }
    suppressHighlighting={true}
  >
    25 rue Robert Latouche, Nice, 06200, Côte D’azur, France
  </Text>
</View>
`;

exports[`ShippingCard should render ShippingCard component with onEdit props 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "elevation": 16,
      "shadowColor": "$boxShadowSecondary",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 4,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 15,
        "paddingLeft": 20,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#242424",
          "fontFamily": "Inter, Helvetica, Arial, sans-serif",
          "fontSize": 18,
          "fontWeight": "bold",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      Bruno Fernandes
    </Text>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      onPress={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      responsible={true}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      testID="edit-icon"
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          clipRule={0}
          d="M4.293 17.686a1.003 1.003 0 01-.29-.799l.38-4.17c.042-.457.244-.888.57-1.213l8.995-8.996c.702-.705 1.976-.67 2.716.069l2.738 2.738h.002c.765.767.796 1.985.067 2.715l-8.996 8.996a1.975 1.975 0 01-1.214.57l-4.17.378a1.003 1.003 0 01-.798-.289zm11.726-9.03l-2.695-2.694 1.947-1.95 2.694 2.695-1.947 1.95zm-6.94 6.946l-2.976.271.264-2.955L11.983 7.3l2.697 2.697-5.6 5.604zM19 21.977c.55 0 1-.45 1-1 0-.549-.45-1-1-1H5c-.548 0-1 .451-1 1 0 .55.452 1 1 1h14z"
          fill={
            {
              "payload": 4280558628,
              "type": 0,
            }
          }
          fillRule={0}
          propList={
            [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      {
        "backgroundColor": "#f0f0f0",
        "height": 1,
        "width": "100%",
      }
    }
  />
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#808080",
        "fontFamily": "Tahoma, Helvetica, Arial, sans-serif",
        "fontSize": 14,
        "fontWeight": "normal",
        "lineHeight": 20,
        "paddingBottom": 18,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 12,
      }
    }
    suppressHighlighting={true}
  >
    25 rue Robert Latouche, Nice, 06200, Côte D’azur, France
  </Text>
</View>
`;
